### DISCUSSION TITLE
No reports received

### DISCUSSION THREAD

**Discussion by StephanSinn** [plausible/analytics#756] [Self-Hosted Support]
Dear plausible Team,

I set up a self-hosted plausible, everything works so far as expected. Except for the reports. I followed this guide and activated all of them: 

https://plausible.io/docs/email-reports/

I used the email adress I also registered at the self-hosted plausible, and to which I also received emails from hello@plausible.local for confirmation. 
I do not receive any email report. Why? 

Thanks for your help, and best regards,
Stephan

**Comment by bhavin192** [comment]
I'm facing this issue as well. I'm able to receive all emails related to password reset, new user. Haven't received any report since I have installed Plausible. I can help with any logs if required. Not sure how to debug this as well. 

The email container does not have any logs apart from password reset emails.
I do have these errors in `plausible_1` container (v1.2.0, [`docker.io/plausible/analytics@sha256:ab22a669cfe20cb54fa0d237f8ff7592793204d16ea5f588ceef4f63639cfc24`](https://hub.docker.com/layers/plausible/analytics/v1.2.0/images/sha256-ab22a669cfe20cb54fa0d237f8ff7592793204d16ea5f588ceef4f63639cfc24)):

```
00:00:29.090 [warn] Expected Elixir.Plausible.Workers.CleanEmailVerificationCodes.perform/2 to return :ok, `{:ok, value}`, or {:error, reason}. Instead
received:

{0, nil}

The job will be considered a success.

17:18:44.882 [error] Clickhousex.Protocol (#PID<0.2534.0>) disconnected: ** (Clickhousex.Error) closed
```

**Comment by ukutaht** [comment]
[Access your postgres](https://github.com/plausible/analytics/discussions/621) DB and see what the contents of `oban_jobs` table is. This table records background jobs and their status.

There are two relevant queues you want to check:
1. `SELECT * FROM oban_jobs WHERE queue='schedule_email_reports';`
2. `SELECT * FROM oban_jobs WHERE queue='send_email_reports';`

Any error reports?

**Reply by bhavin192** [reply]
> 1. `SELECT * FROM oban_jobs WHERE queue='schedule_email_reports';`

There are around 161 rows, all of them don't have any errors.

> 2. `SELECT * FROM oban_jobs WHERE queue='send_email_reports';`

There are only 4 rows, out of 4, one of them has this error.
```
  id  |   state   |       queue        |              worker               |                 args                  |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         errors                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | attempt | max_attempts |        inserted_at         |    scheduled_at     |       attempted_at        |       completed_at       |                     attempted_by                     | discarded_at | priority | tags 
------+-----------+--------------------+-----------------------------------+---------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------+--------------+----------------------------+---------------------+---------------------------+--------------------------+------------------------------------------------------+--------------+----------+------
 2293 | discarded | send_email_reports | Plausible.Workers.SendEmailReport | {"site_id": 1, "interval": "weekly"}  | {"{\"at\": \"2021-03-08T03:30:00.378933Z\", \"error\": \"** (throw) {:permanent_failure, \\\"550 maximum allowed line length is 998 octets, got 1360\\\\r\\\\n\\\"}\\n    (gen_smtp 0.15.0) /app/deps/gen_smtp/src/gen_smtp_client.erl:407: :gen_smtp_client.try_DATA/4\\n    (gen_smtp 0.15.0) /app/deps/gen_smtp/src/gen_smtp_client.erl:262: :gen_smtp_client.send_it/2\\n    (bamboo_smtp 2.1.0) lib/bamboo/adapters/smtp_adapter.ex:77: Bamboo.SMTPAdapter.deliver/2\\n    (bamboo 1.6.0) lib/bamboo/mailer.ex:162: Bamboo.Mailer.deliver_now/4\\n    (plausible 0.0.1) lib/plausible/mailer.ex:6: Plausible.Mailer.send_email/1\\n    (plausible 0.0.1) lib/workers/send_email_report.ex:80: Plausible.Workers.SendEmailReport.send_report/5\\n    (plausible 0.0.1) lib/workers/send_email_report.ex:17: anonymous fn/4 in Plausible.Workers.SendEmailReport.perform/2\\n    (elixir 1.11.3) lib/enum.ex:2193: Enum.\\\"-reduce/3-lists^foldl/2-0-\\\"/3\\n\", \"attempt\": 1}"} |       1 |            1 | 2021-03-01 04:00:31.197562 | 2021-03-08 03:30:00 | 2021-03-08 03:30:00.09232 | 2021-03-08 03:30:00.3789 | {plausible@4dfd4f9463bf,send_email_reports,4epji8ak} |              |        0 | {}
```
And all other rows are like this one.
```
  id  |   state   |       queue        |              worker               |                 args                  | errors | attempt | max_attempts |        inserted_at         |    scheduled_at     |       attempted_at        |       completed_at       |                     attempted_by                     | discarded_at | priority | tags 
------+-----------+--------------------+-----------------------------------+---------------------------------------+--------+---------+--------------+----------------------------+---------------------+---------------------------+--------------------------+------------------------------------------------------+--------------+----------+------
 3329 | scheduled | send_email_reports | Plausible.Workers.SendEmailReport | {"site_id": 1, "interval": "weekly"}  | {}     |       0 |            1 | 2021-03-08 04:00:25.751968 | 2021-03-15 03:30:00 |                           |                          |                                                      |              |        0 | {}
```

**Reply by ukutaht** [reply]
> {:permanent_failure, \\\"550 maximum allowed line length is 998 octets, got 1360\\\\r\\\\n\\\"}

Clearly, this is the root cause of the problem. I created an issue for this: https://github.com/plausible/analytics/issues/827

@StephanSinn Can you confirm you have the same issue? Or is it something different?

**Reply by StephanSinn** [reply]
Hi, sorry for my late response. Yes I can confirm. I obtained exactly the same output as @bhavin192. 

> SELECT * FROM oban_jobs WHERE queue='schedule_email_reports';
No errors, only "completed" states

> SELECT * FROM oban_jobs WHERE queue='send_email_reports';

-```
[ RECORD 1 ]+
id           | 5581
state        | scheduled
queue        | send_email_reports
worker       | Plausible.Workers.SendEmailReport
args         | {"site_id": 1, "interval": "weekly"}
errors       | {}
attempt      | 0
max_attempts | 1
inserted_at  | 2021-03-15 08:00:31.482124
scheduled_at | 2021-03-22 08:00:00
attempted_at | 
completed_at | 
attempted_by | 
discarded_at | 
priority     | 0
tags         | {}
-[ RECORD 2 ]+
id           | 4544
state        | discarded
queue        | send_email_reports
worker       | Plausible.Workers.SendEmailReport
args         | {"site_id": 1, "interval": "weekly"}
errors       | {"{\"at\": \"2021-03-15T08:00:00.677590Z\", \"error\": \"** (throw) {:permanent_failure, \\\"550 maximum allowed line length is 998 oct
ets, got 1360\\\\r\\\\n\\\"}\\n    (gen_smtp 0.15.0) /app/deps/gen_smtp/src/gen_smtp_client.erl:407: :gen_smtp_client.try_DATA/4\\n    (gen_smtp 0.15.
0) /app/deps/gen_smtp/src/gen_smtp_client.erl:262: :gen_smtp_client.send_it/2\\n    (bamboo_smtp 2.1.0) lib/bamboo/adapters/smtp_adapter.ex:77: Bamboo
.SMTPAdapter.deliver/2\\n    (bamboo 1.6.0) lib/bamboo/mailer.ex:162: Bamboo.Mailer.deliver_now/4\\n    (plausible 0.0.1) lib/plausible/mailer.ex:6: P
lausible.Mailer.send_email/1\\n    (plausible 0.0.1) lib/workers/send_email_report.ex:80: Plausible.Workers.SendEmailReport.send_report/5\\n    (plaus
ible 0.0.1) lib/workers/send_email_report.ex:17: anonymous fn/4 in Plausible.Workers.SendEmailReport.perform/2\\n    (elixir 1.11.3) lib/enum.ex:2193:
 Enum.\\\"-reduce/3-lists^foldl/2-0-\\\"/3\\n\", \"attempt\": 1}"}
attempt      | 1
max_attempts | 1
inserted_at  | 2021-03-08 08:00:20.441485
scheduled_at | 2021-03-15 08:00:00
attempted_at | 2021-03-15 08:00:00.538321
completed_at | 2021-03-15 08:00:00.677564
attempted_by | {plausible@4f641f4a915b,send_email_reports,8pkdca3e}
discarded_at | 
priority     | 0
tags         | {}
-[ RECORD 3 ]+
id           | 3507
state        | scheduled
queue        | send_email_reports
worker       | Plausible.Workers.SendEmailReport
args         | {"site_id": 1, "interval": "monthly"}
errors       | {}
attempt      | 0
max_attempts | 1
inserted_at  | 2021-03-01 08:00:09.173448
scheduled_at | 2021-04-01 07:00:00
attempted_at | 
completed_at | 
attempted_by | 
discarded_at | 
priority     | 0
tags         | {}
```

**Comment by bhavin192** [comment]
I was able to do a temporary mitigation for this. Basically I'm asking Exim to allow long lines, but there no guarantee that your mail provider will allow that when Exim sends the mail (Gmail seems to be allowing it).

Create a file with name `ignore_smtp_line_limit` with following content:
```conf
# https://stackoverflow.com/a/55957412/6202405
IGNORE_SMTP_LINE_LENGTH_LIMIT=1
```
Then modify the docker-compose.yaml to have something like this:
```yaml
  mail:
    …
    volumes:
      - ${PWD}/ignore_smtp_line_limit:/etc/exim4/exim4.conf.localmacros
```
Then stop and start the compose stack (haven't tested this with docker-compose, I use podman-compose, I had to do down and up again).
